libbeat/autodiscover/providers/docker/config.go

```diff
@@ -0,0 +1,18 @@
+package docker
```

Comment:
I would call the package path `autodiscovery/provider/docker/*` instead of `providers`. It's messy across the code base but I think recently we tried to call it plural if it has multiple implementations directly in the package itself and singular if it has multiple sub packages with one implementation each.

Reply:
There is something I don't like from that approach. If I'm the top folder of these layout and I find `autodiscover/provider` I don't realize I have several different providers inside. What's the benefit of using singular for this case?

Reply:
If you read the full package name it reads that docker is only 1 provider. But this is definitively personal preference and something @urso and I came up with over time. Not something that would hold back this PR :-)

Reply:
@exekias you sneaked it in ;-)